functions/src/functions/webhooks/auth/twilio-verify.ts

```diff
@@ -0,0 +1,108 @@
+import { getAuth } from 'firebase-admin/auth';
+import { getFirestore } from 'firebase-admin/firestore';
+import { HttpsError, onCall } from 'firebase-functions/v2/https';
+import { Twilio } from 'twilio';
+import { TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN, TWILIO_VERIFY_SERVICE_SID } from '../../../config';
+
+if (!TWILIO_ACCOUNT_SID || !TWILIO_AUTH_TOKEN || !TWILIO_VERIFY_SERVICE_SID) {
+	throw new Error('Missing Twilio environment variables');
+}
+
+const twilioClient = new Twilio(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN);
+
+/**
+ * Cloud function to verify OTP sent by Twilio
+ * If valid, it creates/gets a Firebase user and returns a custom token
+ */
+const verifyOtpFunction = onCall({ maxInstances: 10 }, async (request) => {
+	console.log('Function called with request:', request.data);
+	let { phoneNumber, otp } = request.data;
+
+	// Validate inputs
+	if (!phoneNumber || !otp) {
+		console.log('Missing phone number or OTP');
+		throw new HttpsError('invalid-argument', 'Phone number and OTP are required');
+	}
+
+	try {
+		// Format phone number to E.164 format if not already
+		if (!phoneNumber.startsWith('+')) {
+			phoneNumber = `+${phoneNumber}`;
+		}
+
+		// Basic E.164 validation (should start with + followed by digits only)
+		const phoneRegex = /^\+[1-9]\d{1,14}$/;
+		if (!phoneRegex.test(phoneNumber)) {
+			console.log('Invalid phone number format:', phoneNumber);
+			throw new HttpsError('invalid-argument', 'Phone number must be in valid E.164 format (e.g., +12345678901)');
+		}
+
+		console.log(`Attempting to verify OTP for phone: ${phoneNumber}`);
+		// Verify OTP with Twilio
+		const verification = await twilioClient.verify.v2.services(TWILIO_VERIFY_SERVICE_SID).verificationChecks.create({
+			to: phoneNumber,
+			code: otp,
+		});
+
+		console.log('Twilio verification response:', verification);
+
+		// Check if verification was successful
+		if (verification.status !== 'approved') {
+			console.log('OTP verification failed, status:', verification.status);
+			throw new HttpsError('permission-denied', 'Invalid OTP provided');
+		}
+
+		// OTP is valid, create or get Firebase user
+		const auth = getAuth();
```

Comment:
I would use the two utility classes that we have here:

```ts
const authAdmin = new AuthAdmin();
const firestoreAdmin = new FirestoreAdmin();
```

Reply:
@mkue we tried to replace the const auth = getAuth() with 
const authAdmin = new AuthAdmin();
const auth = authAdmin.auth;

But we got this error:
![image](https://github.com/user-attachments/assets/3805e10a-3b46-4e0a-a880-f4ebfac018a8)

Do you have any idea why?